final_project.py

```diff
@@ -0,0 +1,369 @@
+#
+# Arguments: 
+#   network: the gamer network data structure
+#   user:    a string containing the name of the user
+# 
+# Return: 
+#   A list of all connections the user has.
+#   - If the user has no connections, return an empty list.
+#   - If the user is not in network, return None.
+def get_connections(network, user):
+    if user in network:
+        return network[user][0]
+    
+
+# ----------------------------------------------------------------------------- 
+# get_games_liked(network, user): 
+#   Returns a list of all the games a user likes
+#
+# Arguments: 
+#   network: the gamer network data structure
+#   user:    a string containing the name of the user
+# 
+# Return: 
+#   A list of all games the user likes.
+#   - If the user likes no games, return an empty list.
+#   - If the user is not in network, return None.
+def get_games_liked(network,user):
+    if user in network:
+        return network[user][1]
+
+# ----------------------------------------------------------------------------- 
+# add_connection(network, user_A, user_B): 
+#   Adds a connection from user_A to user_B. Make sure to check that both users 
+#   exist in network.
+# 
+# Arguments: 
+#   network: the gamer network data structure 
+#   user_A:  a string with the name of the user the connection is from
+#   user_B:  a string with the name of the user the connection is to
+#
+# Return: 
+#   The updated network with the new connection added.
+#   - If a connection already exists from user_A to user_B, return network unchanged.
+#   - If user_A or user_B is not in network, return False.
+def add_connection(network, user_A, user_B):
+    if user_A in network and user_B in network:
+        if user_B not in get_connections(network, user_A):
+            
+            network[user_A][0].append(user_B)
+            
+            return network
+        else:
+            pass       
+    else:
+        return False
+# ----------------------------------------------------------------------------- 
+# add_new_user(network, user, games): 
+#   Creates a new user profile and adds that user to the network, along with
+#   any game preferences specified in games. Assume that the user has no 
+#   connections to begin with.
+# 
+# Arguments:
+#   network: the gamer network data structure
+#   user:    a string containing the name of the user to be added to the network
+#   games:   a list of strings containing the user's favorite games, e.g.:
+#            ['Ninja Hamsters', 'Super Mushroom Man', 'Dinosaur Diner']
+#
+# Return: 
+#   The updated network with the new user and game preferences added. The new user 
+#   should have no connections.
+#   - If the user already exists in network, return network *UNCHANGED* (do not change
+#     the user's game preferences)
+def add_new_user(network, user, games):
+   
+    if user not in network:
+        connections =[] #new users don't have any connections
+        network[user] = [connections,games]
```

Comment:
hmm here you're making the value a list instead of tuple. you need to use consistent data types. in python its allowed but in general its not good programming practice